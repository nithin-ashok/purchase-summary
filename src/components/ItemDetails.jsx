import React from "react";
import Chair from "../assets/chair.jpg";

const LineItem = ({children}) => (<span className="line-item">{children}</span>);


const ItemDetails = ({item_name="", afterDisc_Price="", beforeDisc_Price="", quantity=""}) => (<table className="table">
    <tr>
        <td><img className="chair" src={Chair} /></td>
        <td className="text-right">
            <div className="text-left font-small">
                <LineItem>{item_name}</LineItem>
                <LineItem>
                    <div className="font-xsmall">
                        <span className="font-bold"> {`$${afterDisc_Price}`} </span>
                        <span className='align-right'>Qty: {quantity}</span>
                        <span className="line-item text-line-through font-darkgrey font-bold">{`$${beforeDisc_Price}`}</span>
                    </div>
                </LineItem>
            </div>
        </td>
    </tr>
</table>);

export default ItemDetails;
