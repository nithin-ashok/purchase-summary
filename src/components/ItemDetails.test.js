import React from 'react';
import ItemDetails from "./ItemDetails";
import renderer from 'react-test-renderer';

const itemDetails = {
    item_name: "Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
    afterDisc_Price: "99.11",
    beforeDisc_Price: 102.96,
    quantity: 1,
}

it("renderes correctly ItemDetails component", () => {
    const tree = renderer.
    create(<ItemDetails {...itemDetails} />).toJSON();

    expect(tree).toMatchSnapshot();
});