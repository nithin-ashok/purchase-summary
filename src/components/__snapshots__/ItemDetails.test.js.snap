// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderes correctly ItemDetails component 1`] = `
<table
  className="table"
>
  <tr>
    <td>
      <img
        className="chair"
        src="chair.jpg"
      />
    </td>
    <td
      className="text-right"
    >
      <div
        className="text-left font-small"
      >
        <span
          className="line-item"
        >
          Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red
        </span>
        <span
          className="line-item"
        >
          <div
            className="font-xsmall"
          >
            <span
              className="font-bold"
            >
               
              $99.11
               
            </span>
            <span
              className="align-right"
            >
              Qty: 
              1
            </span>
            <span
              className="line-item text-line-through font-darkgrey font-bold"
            >
              $102.96
            </span>
          </div>
        </span>
      </div>
    </td>
  </tr>
</table>
`;
